File: workwatch.py
#!/usr/bin/env python3
import subprocess
from datetime import datetime, timedelta
import sys

p = subprocess.Popen(['journalctl', '--since', 'today', '-o', 'short-iso'], stdout=subprocess.PIPE)

dates = []

next(p.stdout)

for line in p.stdout:
    line = line.decode()
    line = line[:24]
    try:
        date = datetime.strptime(line, "%Y-%m-%dT%H:%M:%S%z" )
    except ValueError:
        continue
    dates.append(date)

pairs = zip(dates, dates[1:])

fmt = "%H:%M:%S"

print("start:  {}".format(dates[0].strftime(fmt)))

delta = 10
if len(sys.argv) > 1:
    delta = int(sys.argv[1])

for pair in pairs:
    if pair[1] - pair[0] > timedelta(minutes=delta):
        time0 = pair[0].strftime(fmt)
        time1 = pair[1].strftime(fmt)
        print("pause:  {} - {}".format(time0, time1))

print("end:    {}".format(dates[-1].strftime(fmt)))

File: workwatch.rb
require 'date'
require 'json'

today = Date.today

def format_time(timestamp)
  Time.at(timestamp).strftime("%H:%M")
end

4.downto 0 do |n|
  date = today - n
  log = `journalctl --since #{date} --until #{date + 1} -o json`.lines.map do |line|
    JSON.parse(line)["__REALTIME_TIMESTAMP"].to_i / (1000 * 1000)
  end

  if log.length == 0
    next
  end

  puts date
  last = log[0]
  start = log[0]
  log.drop(1).each do |record|
    if last + 60 * 10 < record
      puts "#{format_time(start)} - #{format_time(last)}"
      start = record
    end
    last = record
  end
  puts "#{format_time(start)} - #{format_time(last)}"
  puts
end